### PR TITLE
Http server test script

### PR DESCRIPTION
**Pull request by michalrudolf**

### CODE REVIEW

File: utils/test_http_server/http_server.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import sys
+
+#   IF YOU WANT TO ADD A TEST
+#       add test data in server_tests/tests.json with JSON syntax
+#       check if current test implementation will test it properly:
+#           http_server_test_funcs.py > send_request()  - if JSON request, dont add content-type, it is added automatically
+#                                       test_response() 
+
+def main():
+    if test_argv() is 0:
```
Comment (joshymjose) [inline utils/test_http_server/http_server.py]: what does this line do?

File: utils/test_http_server/http_server.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return
+    
+    test.init(sys.argv[1])
+    test.test_loop()
+
+
+def test_argv():
+
+    args_count = len(sys.argv)
```
Comment (joshymjose) [inline utils/test_http_server/http_server.py]: use argparse

File: utils/test_http_server/http_server_test_funcs.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
+import requests
+import logging
+import json
+from datetime import datetime
+import time
+
+# FUTURE TESTS IDEAS:
+# admin valid: '{"connect_ip":"10.24.230.10","connect_key":"01234567890123456789","connect_name":"P3D"}'
```
Comment (joshymjose) [inline utils/test_http_server/http_server_test_funcs.py]: remove from here and add to some notes.txt

File: utils/test_http_server/http_server_test_funcs.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+# global variables
+IP_ADDR = ""
+test_cnt = 0
+tests_off = 0
+test_curr = 0
+json_tests = []
+
+# gcode structures
+G_AUTOHOME = {"command":"G28"}
```
Comment (joshymjose) [inline utils/test_http_server/http_server_test_funcs.py]: why this?

File: utils/test_http_server/http_server_test_funcs.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    except ValueError:
+        test_failed(str(res_dic), "Telemetry parsing to JSON")
+        return
+
+    test_telemetry_response(res_dic)
+
+# testing response from printer to local page
+def test_telemetry_response(response_dic):
+    
+    if 'temp_nozzle' not in response_dic or not isinstance(response_dic['temp_nozzle'], int):
```
Comment (joshymjose) [inline utils/test_http_server/http_server_test_funcs.py:136]: not a good idea to return on first fail, instead it is better to log all fial cases to log file directly than returning thestatus

File: utils/test_http_server/http_server_test_funcs.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    # set up next test
+    if (test_curr + 1) >= test_cnt :
+        test_curr = 0
+        off_test_cnt = 0
+    else:
+        test_curr += 1
+
+def send_request():
+    global test_curr, test_cnt, json_tests, IP_ADDR
+
```
Comment (joshymjose) [inline utils/test_http_server/http_server_test_funcs.py:75]: there is no need to parse content of the test, in the tests are prepared the header and body. here you send it

File: utils/test_http_server/http_server.py (hunk 1 above)
Comment (michalrudolf) [reply utils/test_http_server/http_server.py]: test if you gave the sctipt a proper ip address arg

Comment (joshymjose) [reply utils/test_http_server/http_server.py]: you have to use argparse

**Review by joshymjose — CHANGES_REQUESTED** [review]
need to change the structure of the script, especially how log is done

File: utils/test_http_server/http_server_test_funcs.py (hunk 6 above)
Comment (michalrudolf) [reply utils/test_http_server/http_server_test_funcs.py:75]: I dont understand what do you mean by this. Header is not prepared in tests.json, it has to be parsed from json to use in requests. Body is in tests.json prepared as a json structure and it's only passed in requests...

File: utils/test_http_server/http_server_test_funcs.py (hunk 4 above)
Comment (michalrudolf) [reply utils/test_http_server/http_server_test_funcs.py]: leftover...

File: utils/test_http_server/http_server_test_funcs.py (hunk 5 above)
Comment (michalrudolf) [reply utils/test_http_server/http_server_test_funcs.py:136]: It was on purpose this way.. The idea was that when it encounters an error, it prints whole response message and you can look in the log what is missing (that way only first fail is needed).

I have changed it to tell you (in the log) what is missing

**Review by joshymjose — APPROVED** [review]

**Pull request merged**